include/bluetooth/conn.h

```diff
@@ -392,6 +392,23 @@ struct bt_conn_auth_cb {
 #if defined(CONFIG_BT_BREDR)
 	void (*pincode_entry)(struct bt_conn *conn, bool highsec);
 #endif
+
+	/** @brief notify that pairing process was complete.
+	 *
+	 * This callback notifies the applicaiton that the pairing process
+	 * has been completed.
+	 *
+	 * @param conn Connection object.
+	 * @param bonded pairing is bonded or not.
+	 */
+	void (*pairing_complete)(struct bt_conn *conn, bool bonded);
+
+	/** @brief notify that pairing process has failed.
+	 *
+	 * @param conn Connection object.
+	 * @param err_code error code of failure.
+	 */
+	void (*pairing_failed)(struct bt_conn *conn, int err_code);
```

Comment:
As long as we're not exposing the possible values in a public header, I'd question whether we should have an error parameter at all. Are there any good use cases where the app might want to do different things depending on the error?

Reply:
Okay, maybe it is enough to keep just one callback like before, with a `ret_code` argument which is defined to serval values including:
```
BT_PAIRING_FAILED,
BT_PAIRING_NOT_BONDED,
BT_PAIRING_BONDED
```